KerbalStuff/search.py

```diff
@@ -19,68 +18,64 @@ def weigh_result(result, terms):
         if delta > 100:
-            delta = 100 # Don't penalize for oldness past a certain point
+            delta = 100  # Don't penalize for oldness past a certain point
         score -= delta / 5
-    if result.source_link:
+    if mod.source_link:
         score += 10
-    if (result.created - datetime.now()).days < 30:
+    if (mod.created - datetime.now()).days < 30:
         score += 100
     return score
 
 
 def search_mods(ga, text, page, limit):
     terms = text.split(' ')
     query = db.query(Mod).join(Mod.user).join(Mod.versions).join(Mod.game)
-    filters = list()
+    and_filters = list()
+    or_filters = list()
     for term in terms:
         if term.startswith("ver:"):
-            filters.append(Mod.versions.any(ModVersion.gameversion.has(GameVersion.friendly_version == term[4:])))
+            and_filters.append(Mod.versions.any(ModVersion.gameversion.has(GameVersion.friendly_version == term[4:])))
         elif term.startswith("user:"):
-            filters.append(User.username == term[5:])
+            and_filters.append(User.username == term[5:])
         elif term.startswith("game:"):
-            filters.append(Mod.game_id == int(term[5:]))
+            and_filters.append(Mod.game_id == int(term[5:]))
         elif term.startswith("downloads:>"):
-            filters.append(Mod.download_count > int(term[11:]))
+            and_filters.append(Mod.download_count > int(term[11:]))
         elif term.startswith("downloads:<"):
-            filters.append(Mod.download_count < int(term[11:]))
+            and_filters.append(Mod.download_count < int(term[11:]))
         elif term.startswith("followers:>"):
-            filters.append(Mod.follower_count > int(term[11:]))
+            and_filters.append(Mod.follower_count > int(term[11:]))
         elif term.startswith("followers:<"):
-            filters.append(Mod.follower_count < int(term[11:]))
+            and_filters.append(Mod.follower_count < int(term[11:]))
         else:
-            filters.append(Mod.name.ilike('%' + term + '%'))
-            filters.append(User.username.ilike('%' + term + '%'))
-            filters.append(Mod.short_description.ilike('%' + term + '%'))
+            or_filters.append(Mod.name.ilike('%' + term + '%'))
+            or_filters.append(User.username.ilike('%' + term + '%'))
+            or_filters.append(Mod.short_description.ilike('%' + term + '%'))
     if ga:
         query = query.filter(Mod.game_id == ga.id)
-    query = query.filter(or_(*filters))
+    query = query.filter(and_(*and_filters))
+    query = query.filter(or_(*or_filters))
     query = query.filter(Mod.published == True)
-    query = query.order_by(desc(Mod.follower_count)) # We'll do a more sophisticated narrowing down of this in a moment
-    total = math.ceil(query.count() / limit)
-    if page > total:
-        page = total
+    query = query.order_by(desc(Mod.score))
+
+    total_pages = math.ceil(query.count() / limit)
+    print(page)
```

Comment:
Left behind debug statement?

Reply:
Yes, already removed in local branch shortly after opening the PR, just didn't push it yet ^^